ocpmodels/trainers/base_trainer.py

```diff
@@ -9,40 +9,107 @@
 from ocpmodels.common.utils import (
     build_config,
     plot_histogram,
     save_checkpoint,
     warmup_lr_lambda,
 )
+from ocpmodels.modules.evaluator import Evaluator
 from ocpmodels.modules.normalizer import Normalizer
+from torch.nn.parallel.distributed import DistributedDataParallel
 
 
 @registry.register_trainer("base")
 class BaseTrainer:
-    def __init__(self, args=None, local_rank=0):
-        # defaults.
-        self.device = "cpu"
-        self.is_debug = True
-        self.is_vis = True
+    def __init__(
+        self,
+        task,
+        model,
+        dataset,
+        optimizer,
+        identifier,
+        run_dir=None,
+        is_debug=False,
+        is_vis=False,
+        print_every=100,
+        seed=None,
+        logger="tensorboard",
+        local_rank=0,
+        amp=False,
+        name="base_trainer",
+    ):
+        self.name = name
+
+        if run_dir is None:
+            run_dir = os.getcwd()
+        run_dir = Path(run_dir)
+
+        timestamp = datetime.datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+        if identifier:
+            timestamp += "-{}".format(identifier)
+
+        self.config = {
+            "task": task,
+            "model": model.pop("name"),
+            "model_attributes": model,
+            "optim": optimizer,
+            "logger": logger,
+            "amp": amp,
+            "cmd": {
+                "identifier": identifier,
+                "print_every": print_every,
+                "seed": seed,
+                "timestamp": timestamp,
+                "checkpoint_dir": run_dir / "checkpoints" / timestamp,
+                "results_dir": run_dir / "results" / timestamp,
+                "logs_dir": run_dir / "logs" / logger / timestamp,
+            },
+        }
+        # AMP Scaler
+        self.scaler = torch.cuda.amp.GradScaler() if amp else None
+
+        if isinstance(dataset, list):
+            self.config["dataset"] = dataset[0]
+            if len(dataset) > 1:
+                self.config["val_dataset"] = dataset[1]
+            if len(dataset) > 2:
+                self.config["test_dataset"] = dataset[2]
+        else:
+            self.config["dataset"] = dataset
+
+        if not is_debug and distutils.is_master():
+            os.makedirs(self.config["cmd"]["checkpoint_dir"], exist_ok=True)
+            os.makedirs(self.config["cmd"]["results_dir"], exist_ok=True)
+            os.makedirs(self.config["cmd"]["logs_dir"], exist_ok=True)
+
+        self.is_debug = is_debug
+        self.is_vis = is_vis
+        if torch.cuda.is_available():
+            self.device = local_rank
+        else:
+            self.device = "cpu"
+
+        if distutils.is_master():
+            print(yaml.dump(self.config, default_flow_style=False))
+        self.load()
 
-        # load config.
-        if args is not None:
-            self.load_config_from_yaml_and_cmd(args)
+        self.evaluator = Evaluator(task=name)
```

Comment:
Just flagging that this will throw an error by default unless `name` is overridden.. (I see that it is being set to `s2ef` or `is2re` for the corresponding trainers.)